samples/attested_tls/client/enc/openssl_client.cpp

```diff
@@ -0,0 +1,308 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <errno.h>
+#include <openenclave/enclave.h>
+#include <openssl/bio.h>
+#include <openssl/err.h>
+#include <openssl/pem.h>
+#include <openssl/ssl.h>
+#include <openssl/x509.h>
+#include <openssl/x509_vfy.h>
+#include <string.h>
+#include "../../common/openssl_utility.h"
+
+#include <arpa/inet.h>
+#include <netdb.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+int verify_callback(int preverify_ok, X509_STORE_CTX* ctx);
+
+extern "C"
+{
+    int launch_tls_client(char* server_name, char* server_port);
+};
+
+// This routine conducts a simple HTTP request/response communication with
+// server
+int communicate_with_server(SSL* ssl)
+{
+    unsigned char buf[200];
+    int ret = 1;
+    int error = 0;
+    int len = 0;
+    int bytes_written = 0;
+    int bytes_read = 0;
+
+    // Write an GET request to the server
+    printf(TLS_CLIENT "-----> Write to server:\n");
+    len = snprintf((char*)buf, sizeof(buf) - 1, CLIENT_PAYLOAD);
+    while ((bytes_written = SSL_write(ssl, buf, (size_t)len)) <= 0)
+    {
+        error = SSL_get_error(ssl, bytes_written);
+        if (error == SSL_ERROR_WANT_WRITE)
+            continue;
+        printf(TLS_CLIENT "Failed! SSL_write returned %d\n", error);
+        ret = bytes_written;
+        goto done;
+    }
+
+    printf(TLS_CLIENT "%d bytes written\n", bytes_written);
+
+    // Read the HTTP response from server
+    printf(TLS_CLIENT "<---- Read from server:\n");
+    do
+    {
+        len = sizeof(buf) - 1;
+        memset(buf, 0, sizeof(buf));
```

Comment:
Can you double check if we use the secure version (ie memset_s) in other samples? If yes, we need to use secure CRT functions in this sample too.

Reply:
we are not using memset_s in other samples. The other samples that use memset are file-encryptor and datasealing samples. As this is a quick fix, I could do it for other samples too. Will create a separate PR just to keep other sample changes out of this PR.

Reply:
Created https://github.com/openenclave/openenclave/issues/3760 to not to loose track of it.